daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -540,8 +540,14 @@ func xfsSupported() error {
 		return err // error text is descriptive enough
 	}
 
-	// Check if kernel supports xfs filesystem or not.
-	exec.Command("modprobe", "xfs").Run()
+	mountTarget, err := ioutil.TempDir("", "supportsXFS")
+	if err != nil {
+		return errors.Wrapf(err, "error checking for xfs support")
+	}
+
+	/* The mounting will fail--after the module has been loaded.*/
+	defer os.RemoveAll(mountTarget)
+	unix.Mount("none", mountTarget, "xfs", 0, "")
```

Comment:
needs unmount?

Reply:
Why? As the comment says, the mounting will fail.

Reply:
I think we can check supported filesystems in /proc.

Reply:
> I think we can check supported filesystems in /proc.

Even when the module (if it is one) isn't loaded yet? I doubt it.

Reply:
Ok I get it... so `mount()` calls `request_module` which calls `modprobe`... 🤷‍♂